lib/WP_Auth0_LoginManager.php

```diff
@@ -573,6 +562,31 @@ protected function die_on_login( $msg = '', $code = 0 ) {
 		wp_die( $html );
 	}
 
+	/**
+	 * @param $id_token
+	 * @return object
+	 * @throws WP_Auth0_InvalidIdTokenException
+	 */
+	private function decode_id_token( $id_token ) {
+		$idTokenIss  = 'https://' . $this->a0_options->get( 'domain' ) . '/';
+		$sigVerifier = null;
+		if ( 'RS256' === $this->a0_options->get( 'client_signing_algorithm' ) ) {
+			$jwks        = ( new WP_Auth0_JwksFetcher() )->getKeys();
+			$sigVerifier = new WP_Auth0_AsymmetricVerifier( $jwks );
+		} elseif ( 'HS256' === $this->a0_options->get( 'client_signing_algorithm' ) ) {
+			$sigVerifier = new WP_Auth0_SymmetricVerifier( $this->a0_options->get( 'client_secret' ) );
+		}
```

Comment:
"else" scenario????

Also, you're doing `$this->a0_options->get( 'client_signing_algorithm' )` twice. Seems like it can be extracted. Does it have performance impact?

Reply:
`else` is covered above this block, `$sigVerifier` will be `null` and fail.

It's just pulling from an internal array.